core/storage/couchStorage.go

```diff
@@ -0,0 +1,1801 @@
+	id := createObjectCollectionID(orgID, objectType, objectID)
+	result := couchObject{}
+	if err := store.getOne(id, &result); err != nil {
+		return &Error{fmt.Sprintf("Failed to retrieve object. Error: %s.", err)}
+	}
+	result.Status = common.ObjDeleted
+	result.MetaData.Deleted = true
+	result.LastUpdate = time.Now()
+
+	if err := store.upsertObject(id, result); err != nil {
+		return &Error{fmt.Sprintf("Failed to mark object as deleted. Error: %s.", err)}
+	}
+	return nil
+}
+
+// ActivateObject marks object as active
+func (store *CouchStorage) ActivateObject(orgID string, objectType string, objectID string) common.SyncServiceError {
+	id := createObjectCollectionID(orgID, objectType, objectID)
+	result := couchObject{}
+	if err := store.getOne(id, &result); err != nil {
+		return &Error{fmt.Sprintf("Failed to retrieve object. Error: %s.", err)}
+	}
+	result.MetaData.Inactive = false
+	result.LastUpdate = time.Now()
+
+	if err := store.upsertObject(id, result); err != nil {
+		return &Error{fmt.Sprintf("Failed to mark object as active. Error: %s.", err)}
+	}
+	return nil
+}
+
+// UpdateObjectStatus updates object's status
+func (store *CouchStorage) UpdateObjectStatus(orgID string, objectType string, objectID string, status string) common.SyncServiceError {
+	id := createObjectCollectionID(orgID, objectType, objectID)
+	result := couchObject{}
+	if err := store.getOne(id, &result); err != nil {
+		return &Error{fmt.Sprintf("Failed to retrieve object. Error: %s.", err)}
+	}
+	result.Status = status
+	result.LastUpdate = time.Now()
+
+	if err := store.upsertObject(id, result); err != nil {
+		return &Error{fmt.Sprintf("Failed to update object's status. Error: %s.", err)}
+	}
+	return nil
+}
+
+// RetrieveUpdatedObjects returns the list of all the edge updated objects that are not marked as consumed or received
+// If received is true, return objects marked as received
+func (store *CouchStorage) RetrieveUpdatedObjects(orgID string, objectType string, received bool) ([]common.MetaData, common.SyncServiceError) {
+	result := []couchObject{}
+	var query interface{}
+	if received {
+		query = map[string]interface{}{
+			"selector": map[string]interface{}{
+				"$or": []map[string]interface{}{
+					map[string]interface{}{"status": common.CompletelyReceived},
+					map[string]interface{}{"status": common.ObjReceived},
+					map[string]interface{}{"status": common.ObjDeleted}},
+				"metadata.destinationOrgID": orgID, "metadata.objectType": objectType}}
+	} else {
+		query = map[string]interface{}{
+			"selector": map[string]interface{}{
+				"$or": []map[string]interface{}{
+					map[string]interface{}{"status": common.CompletelyReceived},
+					map[string]interface{}{"status": common.ObjDeleted}},
+				"metadata.destinationOrgID": orgID, "metadata.objectType": objectType}}
+	}
+
+	if err := store.findAll(query, &result); err != nil {
+		return nil, &Error{fmt.Sprintf("Failed to fetch the objects. Error: %s.", err)}
```

Comment:
should it return `nil, nil` if err is notFound? (like the implementation in mongoStorage)